src/compose/ComposeBox.js

```diff
@@ -441,6 +546,68 @@ export default function ComposeBox(props: Props): Node {
     messageInputState,
   ]);
 
+  const doQuoteAndReply = useCallback(
+    async message => {
+      const user = allUsersById.get(message.sender_id);
+      if (!user) {
+        showErrorAlert(_('Quote-and-reply failed'));
+        logging.error('Missing user for sender_id in quote-and-reply', {
+          sender_id: message.sender_id,
+        });
+        return;
+      }
+
+      // TODO: If not already there, re-narrow to `message`'s conversation
+      //   narrow, with getNarrowForReply, and do the quote-and-reply there.
+      //   Discussion:
+      //   https://chat.zulip.org/#narrow/stream/243-mobile-team/topic/.23M1975.20Quote.20and.20reply/near/1455302
+
+      setQuoteAndReplyInProgress(true);
+
+      // Set to match quoting_placeholder in quote_and_reply in
+      // static/js/compose_actions.js in the zulip/zulip repo.
+      const quotingPlaceholder = _('[Quoting…]');
+      insertMessageTextAtCursorPosition(quotingPlaceholder, true);
+
+      let rawContent = undefined;
+      try {
+        // TODO: Give feedback when the server round trip takes longer than
+        //   expected.
+        // TODO: Let the user cancel the request so we don't force a
+        //   quote-and-reply after they've given up and perhaps forgotten
+        //   about it.
+        rawContent = await tryFetch(() =>
+          api.getRawMessageContent(auth, { message_id: message.id }, zulipFeatureLevel),
+        );
+      } catch {
+        showErrorAlert(_('Quote-and-reply failed'));
+        setQuoteAndReplyInProgress(false);
+        return;
+      }
```

Comment:
Let's use a try/finally to handle setting this flag back to false -- so that (a) we get to say it just once instead of several times, and also (b) we reliably set it to false even if there's an exception somewhere we didn't expect one. So:
```js
  setQuoteAndReplyInProgress(true);
  try {

    // … the whole rest of the function after that point …

  } finally {
    setQuoteAndReplyInProgress(false);
  }
```

(Then we'll still want a try/catch for specific expected exceptions, namely here around the network request. But those can just show an error and return, leaving the `finally` around the outside to take care of resetting the flag.)